File: book1_companion/psych_stats_with_python_companion_v0_2/scripts/python/ch10_correlation.py
#!/usr/bin/env python3
from __future__ import annotations

import pandas as pd
from scipy import stats

from common import ROOT, rounded, write_result


def main() -> None:
    data_file = "data/ch10_correlation.csv"
    data = pd.read_csv(ROOT / data_file)
    test = stats.pearsonr(data["study_hours"], data["test_score"])
    fields = {
        "study_hours_mean": rounded(data["study_hours"].mean()),
        "test_score_mean": rounded(data["test_score"].mean()),
        "correlation_r": rounded(test.statistic),
        "degrees_of_freedom": int(len(data) - 2),
        "p_value": rounded(test.pvalue),
        "n": int(len(data)),
    }
    out = write_result("ch10", "Pearson correlation", data_file, fields)
    print(f"wrote {out}")


if __name__ == "__main__":
    main()

File: book1_companion/psych_stats_with_python_companion_v0_2_1/scripts/r/ch10_correlation.R
#!/usr/bin/env Rscript
args <- commandArgs(trailingOnly = FALSE)
script_arg <- args[grep('^--file=', args)][1]
root <- normalizePath(file.path(dirname(sub('^--file=', '', script_arg)), '..', '..'), mustWork = TRUE)
source(file.path(root, "scripts", "r", "common.R"))

d <- read.csv(file.path(root, "data", "ch10_correlation.csv"))
test <- cor.test(d$study_hours, d$test_score, method = "pearson")
write_result(root, "ch10", "Pearson correlation", "data/ch10_correlation.csv", list(
  study_hours_mean = mean(d$study_hours), test_score_mean = mean(d$test_score),
  correlation_r = unname(test$estimate), degrees_of_freedom = unname(test$parameter),
  p_value = test$p.value, n = nrow(d)
))
